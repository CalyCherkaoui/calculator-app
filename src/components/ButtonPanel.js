import React from 'react';
import PropTypes from 'prop-types';
import Button from './Button';

const ButtonPanel = ({ clickHandler }) => {
  const handleClick = (buttonName) => { clickHandler(buttonName); };

  return (
    <div className="buttonsPanel">
      <div className="buttons_row">
        <Button name="AC" key="AC" clickHandler={handleClick} />
        <Button name="+/-" clickHandler={handleClick} />
        <Button name="%" clickHandler={handleClick} />
        <Button name="/" clickHandler={handleClick} />
      </div>
      <div className="buttons_row">
        <Button name="7" clickHandler={handleClick} />
        <Button name="8" clickHandler={handleClick} />
        <Button name="9" clickHandler={handleClick} />
        <Button name="*" clickHandler={handleClick} />
      </div>
      <div className="buttons_row">
        <Button name="4" clickHandler={handleClick} />
        <Button name="5" clickHandler={handleClick} />
        <Button name="6" clickHandler={handleClick} />
        <Button name="-" clickHandler={handleClick} />
      </div>
      <div className="buttons_row">
        <Button name="1" clickHandler={handleClick} />
        <Button name="2" clickHandler={handleClick} />
        <Button name="3" clickHandler={handleClick} />
        <Button name="+" clickHandler={handleClick} />
      </div>
      <div className="buttons_row">
        <Button name="0" clickHandler={handleClick} />
        <Button name="." clickHandler={handleClick} />
        <Button name="=" clickHandler={handleClick} />
      </div>
    </div>
  );
};

ButtonPanel.propTypes = { clickHandler: PropTypes.func.isRequired };

export default ButtonPanel;
