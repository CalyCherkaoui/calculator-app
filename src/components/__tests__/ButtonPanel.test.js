import React from 'react';
import shallow from 'enzyme';
import renderer from 'react-test-renderer';
import ButtonPanel from '../ButtonPanel';
import '@testing-library/jest-dom';

const handleClick = () => ({ total: '0', next: '0', operation: '' });
describe(
  'ButtonPanel component',
  () => {
    // let component;
    // beforeEach(() => { component = shallow(<ButtonPanel clickHandler={clickHandler} />); });
    it(
      'rendrs sucessfully compared to its snapshot',
      () => {
        const component = renderer.create(<ButtonPanel clickHandler={handleClick} />);
        expect(component).toMatchSnapshot();
        // eslint-disable-next-line no-unused-expressions
        // expect(component.find('.buttonsPanel')).toBeTruthy;
        // expect(component).toMatchSnapshot();
      },
    );
  },
);
