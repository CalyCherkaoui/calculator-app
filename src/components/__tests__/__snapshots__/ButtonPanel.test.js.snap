// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonPanel component rendrs sucessfully compared to its snapshot 1`] = `
<div
  className="buttonsPanel"
>
  <div
    className="buttons_row"
  >
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      /
    </button>
  </div>
  <div
    className="buttons_row"
  >
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      *
    </button>
  </div>
  <div
    className="buttons_row"
  >
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
  </div>
  <div
    className="buttons_row"
  >
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
  </div>
  <div
    className="buttons_row"
  >
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;
